osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyAttributes.cs

```diff
@@ -7,8 +7,26 @@ namespace osu.Game.Rulesets.Osu.Difficulty
 {
     public class OsuDifficultyAttributes : DifficultyAttributes
     {
-        public double AimStrain;
-        public double SpeedStrain;
+        public double TapSR;
+        public double TapDiff;
+        public double StreamNoteCount;
+        public double[] MashLevels;
+        public double[] TapSkills;
+
+        public double FingerControlSR;
+        public double FingerControlDiff;
+
+        public double AimSR;
+        public double AimDiff;
+        public double AimHiddenFactor;
+        public double[] ComboTPs;
+        public double[] MissTPs;
+        public double[] MissCounts;
+        public double CheeseNoteCount;
+        public double[] CheeseLevels;
+        public double[] CheeseFactors;
```

Comment:
This is a lot of attributes. Are all of these used in the PP processor/can any of them be removed?

How much data is expected in each one of these?

Reply:
All of them are used in the pp processor. CheeseLevels is the same for every map so it can be removed, but others are necessary.
ComboTPs has 50 elements, MissTPs has 20, MissCounts has 20, CheeseLevels has 11, CheeseFactors has 11.